lib/sql.js

```diff
@@ -689,10 +689,36 @@ SQLConnector.prototype.update = function(model, where, data, options, cb) {
  * @param {Function} cb The callback function
  */
 SQLConnector.prototype._replace = function(model, where, data, options, cb) {
+  var self = this;
   var stmt = this.buildReplace(model, where, data, options);
-  this.execute(stmt.sql, stmt.params, options, cb);
+  this.execute(stmt.sql, stmt.params, options, function(err, info) {
+    if (err) return cb(err);
+    if (!info) {
+      var stmt2 = new ParameterizedSQL('SELECT @@ROWCOUNT FROM ' +
+      self.tableEscaped(model));
+      self.execute(stmt2.sql, stmt2.params, options, function(err, affectedRowsCount) {
+        if (err) return cb(err);
+        if (affectedRowsCount.length === 0) {
+          return cb(errorIdNotFoundForReplace(where.id));
+        } else {
+          cb(null, affectedRowsCount);
+        }
+      });
+    }  else if (info && (info.affectedRows === 0 || info.count === 0)) {
```

Comment:
I think we should fix SQL connectors to return `info.affectedRows` to be consistent with `updateAll` and `deleteAll` methods that already return the same. See the discussion in https://github.com/strongloop/loopback/issues/1167